### ISSUE TITLE
Invalid identifiers should be quoted

### ISSUE THREAD

**Issue by gtflip** [sachinraja/zod-to-ts#12]
I ran into this because I'm parsing JSON that has keys containing the dash ("-") character.

Simple example:
```typescript
const simpleTest = z.object({
   "needs-quotes": z.string(),
});
const testRet = ztt.zodToTs(simpleTest, "Test");
const testAlias = ztt.createTypeAlias(testRet.node, "TestType");
console.log(ztt.printNode(testAlias));
```
The above code will output
```typescript
type TestType = {
    needs-quotes: string;
};
```
...which isn't legal TypeScript.

**Comment by sachinraja** [comment]
Thanks for reporting! I can confirm that this is a bug. I know what the problem is, just not sure how to fix it without quoting every property name. Is there an easy way to detect if a property needs to be quoted or not?

**Comment by gtflip** [comment]
I found [this](https://stackoverflow.com/a/9571440/3918949), which started with the author discovering you could use Unicode in variable names.  They used the symbol for pi as a variable name.  They dug into the spec and wrote an online validator to tell you whether a string could be used unquoted.

I _suspect_ you could usually get away with just quoting it if it starts with a number or contains a symbol that's meaningful syntactically, like ones used as operators (+-=<, etc.), and for dividing parts of the code (like braces, quotes, and parentheses).  The only reason I ran into it is because some other code defined the names.  😃

**Comment by sachinraja** [comment]
Interesting, thanks for the link. I think it makes sense here to play it safer and only allow unquoted names that fit this regex `/[A-Za-z0-9]/g` because it's better to quote unnecessarily than not quote something that should be quoted.

**Comment by gtflip** [comment]
`[A-Za-z_$][A-Za-z0-9_$]+` might be better since underbar and dollar are allowed in unquoted names.  That would probably minimize the unnecessary quoting you'll get in real-world usage.

**Comment by sachinraja** [comment]
Good idea, thanks. I think we can also refine it later if needed.

**Comment by sachinraja** [comment]
This should be fixed in [v1.0.1](https://github.com/sachinraja/zod-to-ts/releases/tag/v1.0.1) now!

**Comment by gtflip** [comment]
Worked for my case.  👍
